include/utilities.h

```diff
@@ -459,4 +459,15 @@ void smart_delete (SmartPointer<TYPE> &sp)
     }
 }
 
+
+template <typename VEC>
+void vector_shift(VEC &in_vec, double a_scalar)
+{
+  for (auto i : in_vec.locally_owned_elements())
```

Comment:
Currently we support C++98. This is C++11. 

We have to choose if this library should expect a C++11 compiler, and if so we have to enforce this at configure time, and check that deal.II has been compiled with DEAL_II_WITH_CXX11. 

Otherwise we cannot ensure that the above will work...